File: app/src/main/java/com/iceteaviet/fastfoodfinder/data/remote/ClientAuth.java
package com.iceteaviet.fastfoodfinder.data.remote;

import com.google.firebase.auth.AuthCredential;
import com.google.firebase.auth.FirebaseUser;
import com.iceteaviet.fastfoodfinder.data.remote.user.model.User;

import io.reactivex.Single;

/**
 * Created by tom on 7/17/18.
 */
public interface ClientAuth {
    String getCurrentUserUid();

    boolean isSignedIn();

    void signOut();

    Single<Boolean> signInWithEmailAndPassword(String email, String password);

    Single<FirebaseUser> signInWithCredential(AuthCredential authCredential);

    User getCurrentUser();

    void setCurrentUser(User user);
}

File: app/src/main/kotlin/com/iceteaviet/fastfoodfinder/data/remote/ClientAuth.kt
package com.iceteaviet.fastfoodfinder.data.remote

import com.google.firebase.auth.AuthCredential
import com.google.firebase.auth.FirebaseUser

import io.reactivex.Single

/**
 * Created by tom on 7/17/18.
 */
interface ClientAuth {
    fun getCurrentUserUid(): String

    fun isSignedIn(): Boolean

    fun signOut()

    fun signInWithEmailAndPassword(email: String, password: String): Single<Boolean>

    fun signInWithCredential(authCredential: AuthCredential): Single<FirebaseUser>
}
